test/e2e/kubelet_stats.go

```diff
@@ -252,12 +263,70 @@ func getOneTimeResourceUsageOnNode(c *client.Client, nodeName string, cpuInterva
 	return usageMap, nil
 }
 
+func getKubeSystemContainersResourceUsage(c *client.Client) (resourceUsagePerContainer, error) {
+	pods, err := c.Pods("kube-system").List(labels.Everything(), fields.Everything())
+	if err != nil {
+		return resourceUsagePerContainer{}, err
+	}
+	nodes, err := c.Nodes().List(labels.Everything(), fields.Everything())
+	if err != nil {
+		return resourceUsagePerContainer{}, err
+	}
+	nameToContainerIDMap := make(map[string]string)
+	containerIds := make([]string, 0)
+	for _, pod := range pods.Items {
+		for _, container := range pod.Status.ContainerStatuses {
+			nameToContainerIDMap[pod.Name+"/"+container.Name] = strings.TrimPrefix(container.ContainerID, "docker:/")
+			containerIds = append(containerIds, strings.TrimPrefix(container.ContainerID, "docker:/"))
+		}
+	}
+
+	mutex := sync.Mutex{}
+	wg := sync.WaitGroup{}
+	wg.Add(len(nodes.Items))
+	idToUsageMap := make(resourceUsagePerContainer, len(containerIds))
+	for _, id := range containerIds {
+		idToUsageMap[id] = nil
```

Comment:
I really think that introducing IdToUsageMap is overcomplication and we should be able to do it in one pass.

The idea is to store "containerIdToName" mapping (instead of nameToContainerID).
With that mapping, you can immediately create the correct resourceUsagePerContainer (in line 303) without additional joining at the end.
And it should be much simpler then.